osu.Game/Rulesets/LegacyBeatmapConversionDifficultyInfo.cs

```diff
@@ -0,0 +1,77 @@
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Rulesets
+{
+    /// <summary>
+    /// A set of properties that are required to facilitate beatmap conversion between legacy rulesets.
+    /// </summary>
+    public class LegacyBeatmapConversionDifficultyInfo : IBeatmapDifficultyInfo
+    {
+        /// <summary>
+        /// Whether the beatmap is made for the target ruleset.
+        /// </summary>
+        public bool IsForTargetRuleset { get; set; }
+
+        /// <summary>
+        /// The beatmap circle size.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// </remarks>
+        public float CircleSize { get; set; }
+
+        /// <summary>
+        /// The beatmap overall difficulty.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// </remarks>
+        public float OverallDifficulty { get; set; }
+
+        /// <summary>
+        /// The count of hitcircles in the beatmap.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// <para>When converting from osu! ruleset beatmaps, this is equivalent to the sum of sliders and spinners in the beatmap.</para>
+        /// </remarks>
+        public int CircleCount { get; set; }
+
+        /// <summary>
+        /// The total count of hitobjects in the beatmap.
+        /// </summary>
+        /// <remarks>
+        /// Only used if <see cref="IsForTargetRuleset"/> is <c>false</c>.
+        /// </remarks>
+        public int TotalObjectCount { get; set; }
+
+        float IBeatmapDifficultyInfo.DrainRate => 0;
+        float IBeatmapDifficultyInfo.ApproachRate => 0;
+        double IBeatmapDifficultyInfo.SliderMultiplier => 0;
+        double IBeatmapDifficultyInfo.SliderTickRate => 0;
+
+        public static LegacyBeatmapConversionDifficultyInfo FromAPIBeatmap(APIBeatmap apiBeatmap) => new LegacyBeatmapConversionDifficultyInfo
+        {
+            IsForTargetRuleset = apiBeatmap.RulesetID == 3,
+            CircleSize = apiBeatmap.CircleSize,
+            OverallDifficulty = apiBeatmap.OverallDifficulty,
+            CircleCount = apiBeatmap.CircleCount,
+            TotalObjectCount = apiBeatmap.SliderCount + apiBeatmap.SpinnerCount + apiBeatmap.CircleCount
+        };
+
+        public static LegacyBeatmapConversionDifficultyInfo FromBeatmap(IBeatmap beatmap) => new LegacyBeatmapConversionDifficultyInfo
+        {
+            IsForTargetRuleset = beatmap.BeatmapInfo.Ruleset.OnlineID == 3,
+            CircleSize = beatmap.Difficulty.CircleSize,
+            OverallDifficulty = beatmap.Difficulty.OverallDifficulty,
+            CircleCount = beatmap.HitObjects.Count(h => h is not IHasDuration),
+            TotalObjectCount = beatmap.HitObjects.Count
+        };
```

Comment:
Whoops. This was originally a more mania-specific struct.

Reply:
The first usage is something that `osu-queue-score-statistics` can make use of. I'm not actually sure how to generalise this better than what I've tried to do here...

Reply:
> The first usage is something that `osu-queue-score-statistics` can make use of.

Right, missed that bit...

If these are to live, then two things:

- `IsForTargetRuleset` most definitely should not hardcode `3`, and instead there should be probably like a `rulesetId` param (probably optional?)
- As mentioned in the other comment [`CircleCount` treatment is super weird](https://github.com/ppy/osu/pull/24685#discussion_r1311248427). I'd say that the inputs to this structure should be all unaltered from the original beatmap pre-conversion, and if something lower down (*cough*mania*cough*) needs that weird convert treatment, then it can do so internally depending on `IsForTargetRuleset`.